socketio/src/client/client.rs

```diff
@@ -308,6 +322,84 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_url_auth_integration() -> Result<()> {
+        fn get_url() -> Url {
+            let timestamp = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_millis();
+            let mut url = crate::test::socket_io_restart_url_auth_server();
+            url.set_query(Some(&format!("timestamp={timestamp}")));
+            url
+        }
+
+        let connect_num = Arc::new(AtomicUsize::new(0));
```

Comment:
Does this really need to wrapped into an `Arc`? Can't we send the atomics across the thread boundary?

Reply:
Yes, we need to access it from multiple threads so it must be behind an `Arc`.